python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -516,16 +612,27 @@ class RandomForestClassifier(Base):
         mod_ptr = <size_t> cuml_model_ptr
         treelite_handle = ctypes.c_void_p(mod_ptr).value
 
-        if fil_sparse_format:
-            storage_type = 'SPARSE'
+        if fil_sparse_format == 'auto':
+            storage_type = fil_sparse_format
         elif not fil_sparse_format:
             storage_type = 'DENSE'
-        elif fil_sparse_format == 'auto':
-            storage_type = fil_sparse_format
+        elif fil_sparse_format:
```

Comment:
how about `fil_sparse_format == True`?
Otherwise, "fil_sparse_format='dense'" will end up in this path because "dense" is still true

Reply:
ohh but we change the value of storage_type and not fil_sparse_format. We check if fil_sparse_format is auto first so auto does not get forced to True.
Also flake8 does not like it when we set a condition as if condi is True,
```
python/cuml/ensemble/randomforestclassifier.pyx:619:32: E712 comparison to True should be 'if cond is True:' or 'if cond:'
```

Reply:
Let's think about what is the expected behavior if fil_sparse_format is passed in as some random string like 'foo'? I think you want it to go to the next branch that raises an exception. Right now that will not happen, it will evaluate as True and go through the `storage_type = fil_sparse_format` branch.

Reply:
ohh okay, i see, that slipped my mind, my bad